packages/frontend/src/features/semanticViewer/store/semanticViewerSlice.ts

```diff
@@ -115,6 +160,41 @@ export const semanticViewerSlice = createSlice({
                 state.selectedMetrics.push(action.payload);
             }
         },
+        setActiveEditorTab: (state, action: PayloadAction<EditorTabs>) => {
+            state.activeEditorTab = action.payload;
+            if (action.payload === EditorTabs.VISUALIZATION) {
+                state.activeSidebarTab = SidebarTabs.VISUALIZATION;
+                if (state.selectedChartType === undefined) {
+                    state.selectedChartType = ChartKind.VERTICAL_BAR;
+                }
+            }
+            if (action.payload === EditorTabs.SQL) {
+                state.activeSidebarTab = SidebarTabs.TABLES;
+            }
+        },
+
+        setSavedChartData: (state, action: PayloadAction<any>) => {
+            console.debug('setSavedChartData state', state, action);
+            throw new Error('Not implemented');
+            /*state.savedSqlChart = action.payload;
+            state.name = action.payload.name;
+            state.description = action.payload.description || '';
+            state.sql = action.payload.sql;
+            state.limit = action.payload.limit || 500;
+            state.selectedChartType =
+                action.payload.config.type || ChartKind.VERTICAL_BAR;*/
+        },
+        setSelectedChartType: (state, action: PayloadAction<ChartKind>) => {
+            state.selectedChartType = action.payload;
+        },
+        setFields: (state, action: PayloadAction<SemanticLayerField[]>) => {
+            const sqlColumns: SqlColumn[] = action.payload.map((field) => ({
+                reference: sanitizeFieldId(field.name),
+                type: field.type as unknown as DimensionType,
```

Comment:
We should either always use `SemanticLayer` types for our slices or we will need to convert `SemanticLayerFieldType` into `DimensionType` here.

Casting it as `Dimension` type will cause certain fields to not show up in the selectors, e.g. if you select a time dimension you cannot set it in the X field of the charts

IMO the "best" solution would be to use the types in `packages/common/src/types/semanticLayer.ts` so we don't have to do a lot of type transformations in the frontend, but it would also be the one that would require the most work atm

Reply:
@almeidabbm I have an PR which has a nice type abstractions on how we manage field states. we can merge this as is and once I rebase you can review the PR.

Reply:
yeah, I had to force this type to make it work with the sql runner transformer, we should use semanticLayerField or build a proper conversion between types.